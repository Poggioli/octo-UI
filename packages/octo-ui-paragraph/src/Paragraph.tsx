import { CSS, forwardRef } from "@octo-ui/core";
import { StyledParagraph } from "./styles";
import { DEFAULT_TAG_PARAGRAPH, ParagraphProps } from "./types";

const Paragraph = forwardRef<typeof StyledParagraph, ParagraphProps>(
  ({ color, ...props }, forwardedRef) => {
    const css: CSS = {
      color: color ?? "$grey11",
      ...props.css,
    };

    return (
      <StyledParagraph
        as={DEFAULT_TAG_PARAGRAPH}
        {...props}
        ref={forwardedRef}
        size={{
          "@initial": "3",
          "@medium": "4",
        }}
        css={{
          lineHeight: "25px",
          ...css,

          "@medium": {
            lineHeight: "27px",
            ...css["@medium"],
          },
        }}
      />
    );
  }
);
Paragraph.toString = () => `.${StyledParagraph.className}`;

export { StyledParagraph };
export { Paragraph };
