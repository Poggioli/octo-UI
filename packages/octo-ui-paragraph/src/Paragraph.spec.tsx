import { render } from "@testing-library/react";
import { Paragraph } from "./Paragraph";

describe("<Paragraph />", () => {
  it(`GIVEN a <Paragraph />
      WHEN screen is small than 900px
      THEN should render a <Paragraph /> using size 3
      AND lineHeight 25px`, () => {
    global.innerWidth = 768;
    const rendered = render(<Paragraph>Paragraph</Paragraph>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Paragraph />
      WHEN screen is equal or greater than 900px
      THEN should render a <Paragraph /> using size 4
      AND lineHeight 27px`, () => {
    global.innerWidth = 900;
    const rendered = render(<Paragraph>Paragraph</Paragraph>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Paragraph />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Paragraph.toString()).toBe(".octo-UI-c-gWerhV");
  });

  describe("color props", () => {
    it(`GIVEN a <Paragraph />
        WHEN color props is not setted
        THEN should render a <Paragraph /> $grey11 as color`, () => {
      const rendered = render(<Paragraph>Paragraph</Paragraph>);
      expect(rendered.container).toMatchSnapshot();
    });

    it(`GIVEN a <Paragraph />
        WHEN color props is $grey10
        THEN should render a <Paragraph /> $grey10 as color`, () => {
      const rendered = render(<Paragraph color="$grey10">Paragraph</Paragraph>);
      expect(rendered.container).toMatchSnapshot();
    });
  });
});
