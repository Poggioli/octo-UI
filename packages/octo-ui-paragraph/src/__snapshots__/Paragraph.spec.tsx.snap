// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> GIVEN a <Paragraph />
      WHEN screen is equal or greater than 900px
      THEN should render a <Paragraph /> using size 4
      AND lineHeight 27px 1`] = `
<div>
  <p
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-htKPem-size-4 octo-UI-c-gWerhV-ikRrLrg-css"
  >
    Paragraph
  </p>
</div>
`;

exports[`<Paragraph /> GIVEN a <Paragraph />
      WHEN screen is small than 900px
      THEN should render a <Paragraph /> using size 3
      AND lineHeight 25px 1`] = `
<div>
  <p
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-htKPem-size-4 octo-UI-c-gWerhV-ikRrLrg-css"
  >
    Paragraph
  </p>
</div>
`;

exports[`<Paragraph /> color props GIVEN a <Paragraph />
        WHEN color props is $grey10
        THEN should render a <Paragraph /> $grey10 as color 1`] = `
<div>
  <p
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-htKPem-size-4 octo-UI-c-gWerhV-ihycQdp-css"
  >
    Paragraph
  </p>
</div>
`;

exports[`<Paragraph /> color props GIVEN a <Paragraph />
        WHEN color props is not setted
        THEN should render a <Paragraph /> $grey11 as color 1`] = `
<div>
  <p
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-htKPem-size-4 octo-UI-c-gWerhV-ikRrLrg-css"
  >
    Paragraph
  </p>
</div>
`;
